test/memory/allocation/align.chpl

```diff
@@ -0,0 +1,2 @@
+var ptr = c_malloc(eltType=int, size=1, alignment=16);
+writeln(ptr : uint % 16);
```

Comment:
Seems like it'd be good to test some more cases here...

Reply:
Any suggestions?

Reply:
You could write a for loop varying alignment and check that the result is 0 mod the alignment

Reply:
Sure one second I'll push soon

Reply:
Done